wellcomecollection/catalogue-pipeline
# set up new autoformat workflow

## What does this change?

Use this [workflow](https://github.com/wellcomecollection/.github/blob/main/.github/workflows/scala_formatting.yml) to run formatting on push
Needs https://github.com/wellcomecollection/aws-account-infrastructure/pull/31 for permissions

## How to test

Pushing to any branch in the repo and watch the format workflow run
Add a badly formatted file in the commit so as to check the "push formatting changes" step as well

## How can we measure success?

The formatting workflow run successfully and is able to push a formatting change commit to the branch

## Have we considered potential risks?

This has been tested in [catalogue-api](https://github.com/wellcomecollection/catalogue-api/pull/804/files)
It runs on feature branches so we can test and validate without affecting the main branch or production systems.
